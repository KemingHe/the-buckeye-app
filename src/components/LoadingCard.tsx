export default function LoadingCard({
  message
}: Readonly<{ message: string }>): JSX.Element {
  return (
    <output aria-live="polite" className="card shadow-lg">
      <div className="card-body flex flex-col justify-center items-center">
        <div className="card-title h-12">
          <span
            aria-hidden={true}
            className="loading loading-spinner loading-lg"
          />
        </div>
        <p aria-label="Loading message" className="text-center">
          {message}
        </p>
      </div>
    </output>
  );
}
