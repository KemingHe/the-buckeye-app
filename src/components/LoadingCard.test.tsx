import { type RenderResult, render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import LoadingCard from "./LoadingCard";

const loadingMessage: string = "This is a loading message.";

describe("LoadingCard component", () => {
  it("renders the loading spinner and message", () => {
    render(<LoadingCard message={loadingMessage} />);

    // Select the loading card by its role.
    const statusElement = screen.getByRole("status");
    expect(statusElement).toBeVisible();

    // Select the loading spinner by its class name.
    const spinnerElement = statusElement.querySelector(
      ".loading.loading-spinner.loading-lg"
    );
    expect(spinnerElement).toBeInTheDocument();

    // Select the message by its text content.
    const messageElement = screen.getByText(loadingMessage);
    expect(messageElement).toBeVisible();
  });

  it("matches the snapshot", () => {
    const { asFragment }: RenderResult = render(
      <LoadingCard message={loadingMessage} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
