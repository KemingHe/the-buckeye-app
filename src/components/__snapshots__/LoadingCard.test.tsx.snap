// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingCard component > matches the snapshot 1`] = `
<DocumentFragment>
  <output
    aria-live="polite"
    class="card shadow-lg"
  >
    <div
      class="card-body flex flex-col justify-center items-center"
    >
      <div
        class="card-title h-12"
      >
        <span
          aria-hidden="true"
          class="loading loading-spinner loading-lg"
        />
      </div>
      <p
        aria-label="Loading message"
        class="text-center"
      >
        This is a loading message.
      </p>
    </div>
  </output>
</DocumentFragment>
`;
